### PR TITLE
Getting Distance from Distance Getter

### PR DESCRIPTION
**Pull request by itsmesp8343**
-Flag to tell whether using distance getter or not -methods to get distance and change boolean
-added to periodic
-changed in prep score how to get distance

### CODE REVIEW

**Review by allison324 — APPROVED** [review]

**Review by tails618 — APPROVED** [review]
Lgtm! Remember to format

File: src/main/java/frc/robot/commands/SpeakerPrepScoreSequence.java [hunk 1]
```diff
@@ -12,14 +12,13 @@
...
  */
 public class SpeakerPrepScoreSequence extends SequentialCommandGroup {
-  /** this is a specified distance from the speaker each time until we do limelight stuff */
-  public final double SPEAKER_SHOOTER_DISTANCE_METERS = 5.0;
 
   public SpeakerPrepScoreSequence(Intake intake, Elevator elevator, Shooter shooter) {
     addRequirements(intake, elevator, shooter);
     addCommands(
-        new InstantCommand(() -> shooter.setShooterDistance(SPEAKER_SHOOTER_DISTANCE_METERS)),
+        new InstantCommand(() -> shooter.setShooterDistance(shooter.getDistanceFromDistanceGetter())),
```
Comment (balghane) [inline src/main/java/frc/robot/commands/SpeakerPrepScoreSequence.java]: Replace this with something that sets `distGetBool` to true?

File: src/main/java/frc/robot/commands/SpeakerPrepScoreSequence.java [hunk 2]
```diff
@@ -12,14 +12,13 @@
...
         new GoHomeSequence(intake, elevator, shooter, true),
-        new InstantCommand(() -> shooter.setShooterMode(ShooterMode.SHOOT)));
+        new InstantCommand(() -> shooter.setShooterMode(ShooterMode.SHOOT)),
+        new InstantCommand(() -> shooter.setDistanceGetterOff()));
```
Comment (balghane) [inline src/main/java/frc/robot/commands/SpeakerPrepScoreSequence.java]: Maybe let's only turn it off after we shoot, not at the end of prep

File: src/main/java/frc/robot/subsystems/shooter/Shooter.java [hunk 3]
```diff
@@ -78,11 +79,23 @@ public enum ShooterMode {
   /** How far are we away from the goal (in meters) */
   private double shooterDistanceMeters = 10.0;
 
-  public Shooter() {
+  private DoubleSupplier distanceGetter;
+
+  private Boolean distGetBool;
```
Comment (balghane) [inline src/main/java/frc/robot/subsystems/shooter/Shooter.java:86]: Can you add a `/** */` style comment about what this results in?

**Review by balghane — CHANGES_REQUESTED** [review]
see comments

**Pull request closed without merging**